### PR TITLE
desy: introduce desy spider

### LINKED ISSUE #133: desy spider
Create the desy spider

### PR DESCRIPTION
**Pull request by spirosdelviniotis**
Signed-off-by: Spiros Delviniotis <spirosdelviniotis@gmail.com>

## Related Issue

Closes #133 

## Checklist:

- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [ ] I linked the related issue(s) in the corresponding commit logs.
- [ ] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [x] I've added any new docs if API/utils methods were added.
- [x] I have updated the existing documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by david-caro** [comment]
This is still missing the part of the media loader that populates the fft_file_paths no?

**Comment by david-caro** [comment]
btw. I want to pair program this with you :)

**Comment by spirosdelviniotis** [comment]
> This is still missing the part of the media loader that populates the fft_file_paths no?

@david-caro the `FftFilesPipeline` populates the `file_paths` at [`item_completed`](https://github.com/inspirehep/hepcrawl/pull/155/files#diff-652863950d134e54674cea83bae54b6fR47) (btw I should rename it to `fft_file_paths`).

-----------

> btw. I want to pair program this with you :)

@david-caro Sure! That would be great! 😄

File: hepcrawl/pipelines.py [hunk 1]
```diff
@@ -50,74 +83,26 @@ def __init__(self):
...
-        self.results_data.append(item)
-        return item
+        hep_item = self._post_enhance_item(item, spider)
+
+        validate(hep_item, 'hep')
+        spider.logger.debug('Validated item by Inspire Schemas.')
+
+        self.results_data.append(hep_item)
+
+        return hep_item
```
Comment (david-caro) [inline hepcrawl/pipelines.py]: s/hep_item/hep_record

File: tests/unit/test_arxiv_all.py [hunk 2]
```diff
@@ -26,35 +27,15 @@ def spider():
...
-
-@pytest.fixture
 def many_results(spider):
     """Return results generator from the arxiv spider. Tricky fields, many
     records.
     """
-    from scrapy.http import TextResponse
+    def _get_processed_item(record, spider):
+        item = pipeline.process_item(record, spider)
+        return item
```
Comment (david-caro) [inline tests/unit/test_arxiv_all.py]: s/item/record

File: tests/unit/test_arxiv_all.py [hunk 3]
```diff
@@ -63,10 +44,10 @@ def many_results(spider):
         )
     )
 
-    assert records
     pipeline = InspireCeleryPushPipeline()
     pipeline.open_spider(spider)
-    return [pipeline.process_item(record, spider) for record in records]
+
+    return [_get_processed_item(parsed_item, spider) for parsed_item in parsed_items]
```
Comment (david-caro) [inline tests/unit/test_arxiv_all.py]: s/_get_processed_item/_get_processed_record/

File: tests/unit/test_arxiv_all.py [hunk 4]
```diff
@@ -26,35 +27,15 @@ def spider():
...
-    return [pipeline.process_item(record, spider) for record in records]
-
-
-@pytest.fixture
 def many_results(spider):
     """Return results generator from the arxiv spider. Tricky fields, many
     records.
     """
-    from scrapy.http import TextResponse
+    def _get_processed_item(record, spider):
```
Comment (david-caro) [inline tests/unit/test_arxiv_all.py]: s/record/item/

File: tests/unit/test_elsevier.py [hunk 5]
```diff
@@ -41,9 +41,11 @@ def record():
         response.meta["xml_url"] = 'elsevier/sample_consyn_record.xml'
         tag = '//%s' % spider.itertag
         nodes = get_node(spider, tag, response)
-        parsed_record = spider.parse_node(response, nodes)
-        assert parsed_record
-        return parsed_record
+
+        parsed_item = spider.parse_node(response, nodes)
+        assert parsed_item
```
Comment (david-caro) [inline tests/unit/test_elsevier.py]: assert parsed_item.record

**Comment by spirosdelviniotis** [comment]
Build fixed! :smile: 

LGTM

**Comment by david-caro** [comment]
Looks like cheating to me ;)

**Pull request merged**